### PR TITLE
Chris's angular book homework

### PR DESCRIPTION
**Pull request by chri-roch12**

### CODE REVIEW

**Comment by chri-roch12** [comment]
Completed status: 2/5 - I hit a big snag in trying to get the individual books to show using BookShow. I think I could have gotten more working but couldnt sort out from there.
Confidence with material: 2/5 - The material makes sense, but I am hitting roadblocks setting up successful routes.

File: public/scripts/app.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+console.log("JS works")
+
```
Comment (wakaspar) [inline public/scripts/app.js:2]: be sure to remember to remove sanity checks!

File: index.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <script src="public/scripts/app.js"></script>
+    <script src="public/scripts/controllers/BooksIndexController.js"></script>
+    <script src="public/scripts/controllers/BooksShowController.js"></script>
+
+  </head>
+  <body>
+    <div class="container">
+      <div class="row">
+        <div class="col-md-8 col-md-offset-2">
+
```
Comment (wakaspar) [inline index.html:25]: I always like to label sections of my code.  What IIIIIISSS this!?

File: public/scripts/controllers/booksIndexController.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
+angular.module("bookApp")
+  .controller("BooksIndexController", BooksIndexController)
+
+BooksIndexController.$inject = ["$http"];
+function BooksIndexController(   $http) {
```
Comment (wakaspar) [inline public/scripts/controllers/booksIndexController.js:6]: remove dead white space by your parameters (____$http)

**Review by wakaspar — COMMENTED** [review]
Your code looks solid, but lacks comments.  I like to label my functions individually and label sections of my html by purpose.  Good job, Chris! 💯

File: index.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE html>
+<html ng-app="bookApp">
+  <head>
+    <meta charset="utf-8">
+    <title>Hey</title>
+
+    <!-- STYLESHEETS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
+    <link rel="stylesheet" href="public/styles/styles.css">
```
Comment (mnfmnfm) [inline index.html:9]: This style of link, that doesn't start with a slash, means that refreshing the page on any of your book detail pages will break.

File: index.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    <!-- ANGULAR SCRIPTS -->
+    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.2/angular.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.2/angular-route.js"></script>
+
+    <!-- APPLICATION SCRIPTS -->
+    <script src="public/scripts/app.js"></script>
+    <script src="public/scripts/controllers/BooksIndexController.js"></script>
+    <script src="public/scripts/controllers/BooksShowController.js"></script>
```
Comment (mnfmnfm) [inline index.html:18]: same issue with no starting slash on this and the previous 2 lines

File: index.html [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  </head>
+  <body>
+    <div class="container">
+      <div class="row">
+        <div class="col-md-8 col-md-offset-2">
+
+          <div class="this will be ng-rep">
```
Comment (mnfmnfm) [inline index.html:26]: Using a classname as a comment is weird.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            <div class="row">
+              <input type="text" class="form-control" placeholder="New Book name" ng-model="BooksIndexCtrl.newBook.title">
```
Comment (mnfmnfm) [inline index.html:28]: You should use labels, not placeholders, to explain what data goes where. See https://www.nngroup.com/articles/form-design-placeholders/

File: public/scripts/app.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+angular.module("bookApp", ["ngRoute"])
+  .config(config);
+
+  config.$inject = ["$routeProvider", "$locationProvider"];
```
Comment (mnfmnfm) [inline public/scripts/app.js:6]: This is indented too much and then pollutes the rest of your file.

File: public/scripts/controllers/booksShowController.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
+
+angular.module("bookApp")
+  .controller("BooksShowController", BooksShowController);
+
+BooksShowController.$inject = ["$http", "$routeParams", "$location"];
+function BooksShowController( $http,   $routeParams,   $location){
```
Comment (mnfmnfm) [inline public/scripts/controllers/booksShowController.js:6]: Use your extra whitespace here to make these actually line up.

File: public/scripts/controllers/booksIndexController.js (hunk 3 above)
Comment (mnfmnfm) [reply public/scripts/controllers/booksIndexController.js:6]: the dead whitespace is fine to make them line up

File: public/scripts/controllers/booksShowController.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  var vm = this;
+  var bookId = $routeParams.id;
+
+  $http({
+    method: "GET",
+    url: "https://super-crud.herokuapp.com/books/" +bookId
```
Comment (mnfmnfm) [inline public/scripts/controllers/booksShowController.js:12]: This should use a template string rather than concatenation.

File: public/scripts/controllers/booksShowController.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    $http({
+      method: 'PUT',
+      url: 'https://super-crud.herokuapp.com/books/' + bookToUpdate._id,
+      data: {
+        title : bookToUpdate.title,
+        author : bookToUpdate.author,
+        image : bookToUpdate.image,
+        releaseDate : bookToUpdate.releaseDate
+      }
+    }).then(onBookUpdateSuccess, onError);
```
Comment (mnfmnfm) [inline public/scripts/controllers/booksShowController.js:33]: You don't actually have a function called onError, do you? :(

File: templates/books-show.html [hunk 12]
```diff
@@ -0,0 +1,20 @@
+<div ng-model="book in booksShowCtrl" class="container">
```
Comment (mnfmnfm) [inline templates/books-show.html:1]: The "in" syntax is usually used for ng-repeat, not for ng-model, and I don't think you need any ng-model on this element.

File: templates/books-show.html [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    <div class="col-sm-6">
+      <label ng-click="editing = !editing"><h4 class="bold">Author:</h4><h4 ng-class="editing-label" class="underline">{{booksShowCtrl.book.author}}</h4></label>
+      <input ng-show="editing">
+      <label ng-click="editing = !editing"><h4 class="bold">Release Date:</h4><h4 ng-class="editing-label" class="underline">{{booksShowCtrl.book.releaseDate}}</h4></label>
+      <input ng-show="editing">
+      <p>
+        Click on an item to edit.
+      </p>
+      <div class="buttons">
+        <a href="" ng-click="">Save changes</a>
```
Comment (mnfmnfm) [inline templates/books-show.html:16]: If you don't have an href or an ng-click, you don't need those attributes. I wish you'd been able to get this working!

**Review by mnfmnfm — COMMENTED** [review]
It looks like you didn't quite get through this assignment, which concerns me. Your single-book page doesn't seem to be working beyond the absolute basics.
